packages/in_app_purchase/in_app_purchase_android/CHANGELOG.md

```diff
@@ -1,3 +1,13 @@
+## 0.4.0
+
+* Updates Google Play Billing Library from 6.2.0 to 7.1.1.
+* **BREAKING CHANGES**:
+  * Removes the deprecated `ProrationMode` enum. `ReplacementMode` should be used instead.
+  * Removes the deprecated `BillingClientWrapper.enablePendingPurchases` method.
+  * Bumps `minSdk` to 21.
```

Comment:
This would be a breaking change for `in_app_purchase` when rolling the dependency there; we strongly prefer not dropping OS support before the SDK itself.

In this case, Flutter 3.24+ only supports Android SDK 21+, so the way to make this change would be to update the plugin's min SDK version to 3.24, which makes this no longer a breaking change.